### PR TITLE
feat: Adding Decorative + Shadow Colors / Fix Background.discount

### PR DESCRIPTION
**Pull request by baptistedajon**
## Changes description
Adding Decorative + Shadow Colors / Fix Background.discount

## Context
Needed new colors for my project

## Checklist
- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on an iPhone device/simulator.
- [x] I have tested on an iPad device/simulator.

## Does this introduce a breaking change?
- No

### CODE REVIEW

File: Sources/Vitamin/Foundations/Colors/Semantic/Colors+Theme.swift [hunk 1]
```diff
@@ -53,7 +65,7 @@ extension VitaminColor {
                 public static let backgroundBrandPrimary = VitaminColor.Base.blue300.color
                 public static let backgroundBrandSecondary = VitaminColor.Base.blue700.color
                 public static let backgroundAccent = VitaminColor.Base.yellow400.color
-                public static let backgroundDiscount = VitaminColor.Base.red400.color
+                public static let backgroundAlert = VitaminColor.Base.red400.color
```
Comment (daniel-dumortier) [inline Sources/Vitamin/Foundations/Colors/Semantic/Colors+Theme.swift:68]: Same than above

**Review by daniel-dumortier — CHANGES_REQUESTED** [review]
Great job ! 👍
I made just two little remarks : 
- I think the `backgroundDiscount` in Core and Light Themes should be kept as deprecated (they are public, even if their use is discouraged)
- I think the new `Shadow` semantic color should be used in the Shadow foundation

Since I made some of the changes also in another PR (for Tag implementation) because I needed them, I think that could be great if we merge this one first, then I'll merge main in my branch to resolve conflicts before merging it

File: Sources/Vitamin/Foundations/Colors/Semantic/Colors+Theme.swift [hunk 2]
```diff
@@ -15,6 +15,7 @@ extension VitaminColor {
                 public static let backgroundBrandPrimary = VitaminColor.Base.blue400.color
                 public static let backgroundBrandSecondary = VitaminColor.Base.blue50.color
                 public static let backgroundAccent = VitaminColor.Base.yellow400.color
+                public static let backgroundDiscount = VitaminColor.Base.red400.color
```
Comment (florentlotthepro) [inline Sources/Vitamin/Foundations/Colors/Semantic/Colors+Theme.swift:18]: `@available(*, deprecated, renamed: "backgroundAlert")`

File: Sources/Vitamin/Foundations/Colors/Semantic/Colors+Theme.swift [hunk 3]
```diff
@@ -65,6 +66,7 @@ extension VitaminColor {
                 public static let backgroundBrandPrimary = VitaminColor.Base.blue300.color
                 public static let backgroundBrandSecondary = VitaminColor.Base.blue700.color
                 public static let backgroundAccent = VitaminColor.Base.yellow400.color
+                public static let backgroundDiscount = VitaminColor.Base.red400.color
```
Comment (florentlotthepro) [inline Sources/Vitamin/Foundations/Colors/Semantic/Colors+Theme.swift:69]: `@available(*, deprecated, renamed: "backgroundAlert")`

**Review by florentlotthepro — APPROVED** [review]
I've added some suggestions for deprecated warning.

**Comment by lauthieb** [comment]
@baptistedajon feel free to fix conflicts.
Can you please take this opportunity by taking this issue: https://github.com/Decathlon/vitamin-ios/issues/23?
Thanks

File: Sources/Vitamin/Foundations/Colors/Semantic/Colors+Shadow.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
+//
+//  Vitamin iOS
+//  Apache License 2.0
+//
+
+import UIKit
+
+extension VitaminColor.Core {
+    public enum Shadow {
```
[the remark this answers is not in the document]
Comment (daniel-dumortier) [reply Sources/Vitamin/Foundations/Colors/Semantic/Colors+Shadow.swift:9]: fixed

**Comment by daniel-dumortier** [comment]
@lauthieb : I made the last changes, resolved the conflicts, and integrated the missing refacto from #23 (some of them had already been integrated before)
I also linked the corresponding issues to this PR

Whenever @florentlotthepro or @baptistedajon approves, I will be able to merge it

@florentlotthepro : I did not write any new comment, there should be no typos this time ;)

**Pull request merged**